src/services/burnerWalletService.ts

```diff
@@ -4,74 +4,58 @@ import { ethers, Wallet } from 'ethers'
 
-export const createSignerAndWallet = async (API: API): Promise<Wallet> => {
-  const newSigner = ethers.Wallet.createRandom()
-  const walletAddress = await API.initWallet({
-    ownerAddress: newSigner.address
-  })
-  window.localStorage.setItem(
-    `cometh-connect-${walletAddress}`,
-    newSigner.privateKey
-  )
-
-  return newSigner
+export const _setSignerLocalStorage = (
+  identifier: string,
+  privateKey: string
+): void => {
+  window.localStorage.setItem(`cometh-connect-${identifier}`, privateKey)
 }
 
-export const createSignerAndWalletForUserId = async (
-  token: string,
-  userId: string,
-  API: API
-): Promise<Wallet> => {
-  const newSigner = ethers.Wallet.createRandom()
-  window.localStorage.setItem(`cometh-connect-${userId}`, newSigner.privateKey)
-
-  await API.initWalletForUserID({
-    token,
-    ownerAddress: newSigner.address
-  })
-
-  return newSigner
+export const _getSignerLocalStorage = (identifier: string): string | null => {
+  return window.localStorage.getItem(`cometh-connect-${identifier}`)
 }
 
-export const getSigner = async (
-  API: API,
-  provider: StaticJsonRpcProvider,
-  walletAddress: string
-): Promise<Wallet> => {
-  const storagePrivateKey = window.localStorage.getItem(
-    `cometh-connect-${walletAddress}`
-  )
-  if (!storagePrivateKey)
-    throw new Error(
-      'New Domain detected. You need to add that domain as signer.'
-    )
+export const createSigner = async ({
+  API,
+  walletAddress,
+  token,
+  userId
+}: {
+  API: API
+  walletAddress?: string
+  token?: string
+  userId?: string
+}): Promise<{ signer: Wallet; walletAddress: string }> => {
+  const signer = ethers.Wallet.createRandom()
 
-  const storageSigner = new ethers.Wallet(storagePrivateKey)
+  // if import external safe wallet
+  if (walletAddress) {
+    _setSignerLocalStorage(walletAddress, signer.privateKey)
+    return { signer, walletAddress }
+  }
 
-  const isOwner = await safeService.isSigner(
-    storageSigner.address,
-    walletAddress,
-    provider,
-    API
-  )
+  // if safe created by cometh wallet SDK
+  const predictedWalletAddress = await API.getWalletAddress(signer.address)
 
-  if (!isOwner)
-    throw new Error(
-      'New Domain detected. You need to add that domain as signer.'
-    )
+  const identifier = userId ? userId : predictedWalletAddress
+  _setSignerLocalStorage(identifier, signer.privateKey)
 
-  return storageSigner
+  return { signer, walletAddress: predictedWalletAddress }
 }
 
-export const getSignerForUserId = async (
-  userId: string,
-  API: API,
-  provider: StaticJsonRpcProvider,
+export const getSigner = async ({
+  API,
+  provider,
+  walletAddress,
+  userId
+}: {
+  API: API
+  provider: StaticJsonRpcProvider
   walletAddress: string
-): Promise<Wallet> => {
-  const storagePrivateKey = window.localStorage.getItem(
-    `cometh-connect-${userId}`
-  )
+  userId?: string
+}): Promise<Wallet> => {
+  const identifier = userId ? userId : walletAddress
```

Comment:
Same here, this should not be manage by the caller. The caller know he have a wallet address or  a userID. Here just manage identifier.